EIPS/eip-712.md

```diff
@@ -414,13 +414,39 @@ To be done before this EIP can be considered accepted:
 *   [x] Finalize specification of structure hashing
 *   [x] Domain separators
 *   [x] Add test vectors
-*   [ ] Review specification
+*   [x] Review specification
 
 To be done before this EIP can be considered "Final":
 
-*   [ ] Implement `eth_signTypedData` in major RPC providers.
-*   [ ] Implement `web3.eth.signTypedData` in Web3 providers.
-*   [ ] Implement `keccak256` struct hashing in Solidity.
+*   [x] Implement `eth_signTypedData` in major RPC providers.
+*   [x] Implement `web3.eth.signTypedData` in Web3 providers.
+*   [x] Implement `keccak256` struct hashing in Solidity.
+
+**Rationale for marking goals as done:**
+* `eth_signTypedData`
+
+MetaMask has had a series of `eth_signTypedData` since 2017 ([article](https://medium.com/metamask/scaling-web3-with-signtypeddata-91d6efc8b290)), and is an accepted standard in the wallet ecosystem.
+
+* `web3.eth.signTypedData`
+
+While this api does **not** exist in the web3.js library as of this writing, it is included as an experimnetal feature in the Ethers library [here](https://medium.com/metamask/scaling-web3-with-signtypeddata-91d6efc8b290). It can also be called from the Web3.js library by using the `sendAsync` method:
+
+```javascript
+web3.currentProvider.sendAsync({
+  method: 'eth_signTypedData',
+  params: [msgParams, from]
+```
+In addition, the `signTypedData` function is of limited utility in web3 convenience libraries, as it is generally relevant to the interface that stores the private keys used for signing, and with the excepetionof burner wallets, this is almost always done in practice within a wallet.
+
+* `keccak256` struct hashing
+
+While the `keccak256` function in Solidity cannot be invoked directly on structs, the introduction of `abi.encode` gives a sufficient and well-accepted avenue to hashing structs.
```

Comment:
And when gas cost matters, dropping to assembly is always an option.

Reply:
Thanks for taking a look! Do you think this needs to be included in the text, or should it be left as is?